firehose/firehose.go

```diff
@@ -112,29 +108,39 @@ func NewOutputPlugin(region, deliveryStream, dataKeys, roleARN, endpoint, timeKe
 	}, nil
 }
 
-func newPutRecordBatcher(roleARN string, sess *session.Session, endpoint string) *firehose.Firehose {
-	svcConfig := &aws.Config{}
-	if endpoint != "" {
-		defaultResolver := endpoints.DefaultResolver()
-		cwCustomResolverFn := func(service, region string, optFns ...func(*endpoints.Options)) (endpoints.ResolvedEndpoint, error) {
-			if service == "firehose" {
-				return endpoints.ResolvedEndpoint{
-					URL: endpoint,
-				}, nil
-			}
-			return defaultResolver.EndpointFor(service, region, optFns...)
+func newPutRecordBatcher(roleARN, region, firehoseEndpoint, stsEndpoint string) (*firehose.Firehose, error) {
+	defaultResolver := endpoints.DefaultResolver()
```

Comment:
Is there any specific reason we're referencing `defaultResolver` via a closure, or can we create this inside of the `customResolverFn` function definition?